MAINTAINERS.md

```diff
@@ -0,0 +1,34 @@
+# Maintainers
+
+This page includes information for maintainers of this repo.
+
+## How to manually generate the binaries for a Github release
+
+The binaries for the Github release are generated using [GoReleaser](https://goreleaser.com/). The Github workflow [ci-release.yml](./.github/workflows/ci-release.yml) should automatically create pre-build binaries and attach them to the release. However, if the Github workflow fails, you can manually generate pre-built binaries and attach them to the release.
+
+> **NOTE** Due to `goreleaser`'s CGO limitations, cross-compiling the binary does not work. So the binaries must be built on the target platform. This means that the release process must be done on a Linux amd64 machine.
+
+To generate the binaries for the Github release, you can run the following command:
+
+```sh
+make goreleaser-release
+```
+
+This will generate the binaries as defined in `.goreleaser.yaml` and put them in `build/goreleaser` like so:
+
+```sh
+build
+└── goreleaser
+    ├── CHANGELOG.md
+    ├── artifacts.json
+    ├── celestia-app_Linux_x86_64.tar.gz
+    ├── celestia-app_linux_amd64_v1
+    │   └── celestia-appd
+    ├── checksums.txt
+    ├── config.yaml
+    └── metadata.json
+```
+
+For the Github release, you just need to upload the `checksums.txt` and `celestia-app_Linux_x86_64.tar.gz` files.
+
+> **NOTE** Since you built the binaries manually, the checksums.txt won't automatically be signed with the GPG key associated with celestia-app-maintainers so there won't be a `checksums.txt.sig` to upload.
```

Comment:
```suggestion
> **NOTE** If you are generating and signing the release binaries, your public key must be one of the available keys for verification.
```

If we are implementing signed binaries, then all releases should be signed binaries. 
The celestia-app-maintainers key can be the default used in the github actions, but as a fall back, there should be other team members that have published public keys for verification. 

This means we should add steps here for setting up a gpg key to sign the binaries along side updating the `make release` to sign the binaries. 

Then for any team member that plans to publish a release, they would need to add their public key to the list of available keys. 

For verification in this scenario, we might just need to make a simple bash script that loops the public keys and checks each one. As long as one is valid then the release is valid.

Reply:
This addresses your questions posed in the PR description. We shouldn't share the gpg key, we should just have multiple signers.

Reply:
Resolved in a bunch of subsequent commits on this PR.